Framework/Kernel/src/ConfigService.cpp

```diff
@@ -1982,22 +1980,28 @@ ConfigServiceImpl::getFacility(const std::string &facilityName) const {
  * @throw NotFoundException if the facility is not found
  */
 void ConfigServiceImpl::setFacility(const std::string &facilityName) {
-  bool found = false;
-  // Look through the facilities for a matching one.
-  std::vector<FacilityInfo *>::const_iterator it = m_facilities.begin();
-  for (; it != m_facilities.end(); ++it) {
-    if ((**it).name() == facilityName) {
-      // Found the facility
-      found = true;
-      // So it's safe to set it as our default
-      setString("default.facility", facilityName);
-    }
-  }
-  if (!found) {
+  const FacilityInfo *foundFacility = nullptr;
+
+  try {
+    // Get facility looks up by string - so re-use that to check if the facility
+    // is known
+    foundFacility = &getFacility(facilityName);
+  } catch (const Exception::NotFoundError &) {
     g_log.error("Failed to set default facility to be " + facilityName +
                 ". Facility not found");
-    throw Exception::NotFoundError("Facilities", facilityName);
+    throw;
+  }
+  assert(foundFacility);
+  setString("default.facility", facilityName);
+
+  const auto &associatedInsts = foundFacility->instruments();
+  if (associatedInsts.size() == 0) {
```

Comment:
Can we use `empty()` here instead?
I'm wondering what needs to be done in this case, maybe one should actually throw instead of just logging? Otherwise, what will be the value of the `default.instrument` ?

Reply:
Good catch
I've added a step to throw if there are no instruments, as I suspect that people will start to rely on this without checking that the chosen facility has correctly set an instrument (however unlikely) so throwing should alert us something screwy is happening